### PR TITLE
fix checkout bug where pages/checkout.js redirects to homepage on successful checkout

### PR DESCRIPTION
**Pull request by john-raymon**

### CODE REVIEW

File: pages/checkout/index.js [hunk 1]
```diff
@@ -533,6 +532,10 @@ class CheckoutPage extends Component {
 }
 
 CheckoutPage.propTypes = {
+  orderReceipt: PropTypes.oneOfType([
+    PropTypes.object,
+    null
```
Comment (john-raymon) [inline pages/checkout/index.js:537]: @ScopeyNZ I might have to change this

File: components/checkout/Confirm.js [hunk 2]
```diff
@@ -35,7 +42,7 @@ class Confirm extends Component {
   render() {
     const { orderReceipt } = this.props;
 
-    return (
+    return !orderReceipt ? '' : (
```
Comment (robbieaverill) [inline components/checkout/Confirm.js:45]:
```suggestion
    if (!orderReceipt) {
      return '';
    }

    return (
```

File: components/common/atoms/Dropdown.js [hunk 3]
```diff
@@ -23,12 +23,13 @@ class Dropdown extends Component {
         >
           <p className="font-color-medium">{this.props.value || this.props.placeholder }</p>
           <img
+            alt="downward-arrow-symbol"
```
Comment (robbieaverill) [inline components/common/atoms/Dropdown.js:26]: This doesn't really provide much additional context as an alt tag, can you use something like "Downward symbol indicating opening of a dropdown" or something?

File: pages/checkout/index.js [hunk 4]
```diff
@@ -533,6 +530,10 @@ class CheckoutPage extends Component {
 }
 
 CheckoutPage.propTypes = {
+  orderReceipt: PropTypes.oneOfType([
+    PropTypes.object,
+    PropTypes.oneOf([null]),
+  ]),
```
Comment (robbieaverill) [inline pages/checkout/index.js:536]:
```suggestion
  orderReceipt: PropTypes.oneOfType([
    PropTypes.object,
    null,
  ]),
```

Why would it be passed but be null?

File: components/common/atoms/Dropdown.js (hunk 3 above)
Comment (john-raymon) [reply components/common/atoms/Dropdown.js:26]:
```suggestion
            alt="Downward symbol indicating opening of a dropdown"
```

File: pages/checkout/index.js (hunk 4 above)
Comment (john-raymon) [reply pages/checkout/index.js]: If you're asking why do we pass `null`, it's to know not to render the checkout/confirm page and also not to regenerate a checkoutToken here https://github.com/chec/commercejs-nextjs-demo-store/pull/90/files#diff-295c5e85a1a560e6f37f2ae1d68ab5e8R93.

Looking at it again, I'm not sure it's the best approach.

Comment (john-raymon) [reply pages/checkout/index.js]: Going to revert your suggestion as React's throwing this error now
`Warning: Invalid argument supplied to oneOfType. Expected an array of check functions, but received null at index 1.`

**Pull request merged**